packages/pretty-format/src/plugins/dom_collection.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Config, NewPlugin, Printer, Refs} from 'types/PrettyFormat';
+
+import {printObjectProperties} from '../collections';
+
+const SPACE = ' ';
+
+const COLLECTION_NAMES = ['DOMStringMap', 'NamedNodeMap'];
+
+export const test = (val: any) =>
+  val &&
+  val.constructor &&
+  COLLECTION_NAMES.indexOf(val.constructor.name) !== -1;
+
+const convertCollectionToObject = (collection: any) => {
+  let result = {};
+
+  if (collection.constructor.name === 'NamedNodeMap') {
+    for (let i = 0; i < collection.length; i++) {
+      result[collection[i].name] = collection[i].value;
+    }
+  } else {
+    result = Object.assign({}, collection);
+  }
+
+  return result;
+};
+
+export const serialize = (
+  collection: any,
+  config: Config,
+  indentation: string,
+  depth: number,
+  refs: Refs,
+  printer: Printer,
+): string =>
+  collection.constructor.name +
```

Comment:
A possible improvement which you can decide whether to make, or not, is implement the depth limit:

```js
++depth > config.maxdepth
  ? '[' + collection.constructor.name + ']'
  : /* existing expression */
```

For example:
* https://github.com/facebook/jest/blob/master/packages/pretty-format/src/index.js#L186-L244
* https://github.com/facebook/jest/blob/master/packages/pretty-format/src/plugins/dom_element.js#L82-L84
* https://github.com/facebook/jest/blob/master/packages/pretty-format/src/plugins/immutable.js#L29-L51

Reply:
I think depth limit is not needed for DOM collections, because usually it's flat, but I can do it if needed.

Reply:
Agree, let’s keep it short and simple.

You are so quick with this super PR that I have not had time to test hypothesis that the problem is extra props in jsdom implementation and that pretty-format could serialize real DOM collections in browser. If we find problems with any other data structures, we might consider whether there is a correct and efficient test to detect jsdom objects more generically.